safe/gui/tools/multi_exposure_dialog.py

```diff
@@ -0,0 +1,179 @@
+            try:
+                title = self.keyword_io.read_keywords(layer, 'title')
+            except (NoKeywordsFoundError,
+                    KeywordNotFoundError, MetadataReadError):
+                # Skip if there are no keywords at all, or missing keyword
+                continue
+            except:  # pylint: disable=W0702
+                pass
+            else:
+                # Lookup internationalised title if available
+                title = self.tr(title)
+
+            # Register title with layer
+            set_layer_from_title = setting(
+                'set_layer_from_title_flag', True, bool)
+            if title and set_layer_from_title:
+                if qgis_version() >= 21800:
+                    layer.setName(title)
+                else:
+                    # QGIS 2.14
+                    layer.setLayerName(title)
+
+            source = layer.id()
+
+            if layer_purpose == layer_purpose_hazard['key']:
+                add_ordered_combo_item(
+                    self.cbx_hazard, title, source)
+            elif layer_purpose == layer_purpose_aggregation['key']:
+                add_ordered_combo_item(
+                    self.cbx_aggregation, title, source)
+            elif layer_purpose == layer_purpose_exposure['key']:
+
+                # fetching the exposure
+                try:
+                    exposure_type = self.keyword_io.read_keywords(
+                        layer, layer_purpose_exposure['key'])
+                except:  # pylint: disable=W0702
+                    # continue ignoring this layer
+                    continue
+
+                for key, combo in self.combos_exposures.iteritems():
+                    if key == exposure_type:
+                        add_ordered_combo_item(
+                            combo, title, source)
+
+        self.cbx_hazard.setEnabled(self.cbx_hazard.count())
+        self.cbx_aggregation.setEnabled(self.cbx_aggregation.count())
+        for combo in self.combos_exposures.itervalues():
+            combo.setEnabled(combo.count())
+
+    def accept(self):
+        """Launch the multi exposure analysis."""
+        hazard = layer_from_combo(self.cbx_hazard)
+        aggregation = layer_from_combo(self.cbx_aggregation)
+        exposures = []
+        for combo in self.combos_exposures.itervalues():
+            exposures.append(layer_from_combo(combo))
+
+        impact_function = MultiExposureImpactFunction()
+        impact_function.hazard = hazard
+        impact_function.exposures = exposures
+        impact_function.aggregation = aggregation
+
+        code, message = impact_function.prepare()
+        # self.assertEqual(code, PREPARE_MULTI_SUCCESS, message)
+
+        code, message = impact_function.run()
+        # self.assertEqual(code, ANALYSIS_MULTI_SUCCESS, message)
+
+        for analysis in impact_function.impact_functions():
+            add_impact_layers_to_canvas(self.impact_function, iface)
+
+
+
```

Comment:
__[flake8]__
* [W391] blank line at end of file

<!-- Comment by [SideCI](https://sideci.com) -->